### PR TITLE
Zingo sync

### PR DESCRIPTION
**Pull request by Oscar-Pepper**

### CODE REVIEW

File: zingo-sync/src/map_nullifier.rs
Comment (fluidvanadium) [inline zingo-sync/src/map_nullifier.rs:1]: according to Kris, there are too many nullifiers to store all of them

Comment (fluidvanadium) [inline zingo-sync/src/map_nullifier.rs:1]: according to Kris, there are too many nullifiers to store all of them.
we need to only store those nullifiers we know are relevant

File: zingo-sync/src/scan_director.rs [hunk 1]
```diff
@@ -0,0 +1,52 @@
+//! Direct all scanning operations.
+//!
+//! The entry point for a lightclient synchronization process.
+//! This module has a higher complexity and will likely be formed of many sub-modules.
+//!
+//! Takes the wallet's keys, wallet options and birthday as input.
```
Comment (fluidvanadium) [inline zingo-sync/src/scan_director.rs:6]: yes

File: zingo-sync/src/scan_director.rs [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+//!
+//! Load from wallet file (if exists) and prepare/restore scanning range:
```
Comment (fluidvanadium) [inline zingo-sync/src/scan_director.rs:8]: there is a spanning tree for scanning ranges implemented here https://github.com/zcash/librustzcash/blob/45df8b785313a708fed6ed47e30bc201a11a5c99/zcash_client_backend/src/data_api/scanning.rs#L7

File: zingo-sync/src/scan_director.rs [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+//!
+//! Transaction scanning:
+//! - When the trial decryptor, transparent TXID fetcher or a match from nullifier map returns a
+//! TXID, add to the queued transactions list and pass to `priority` to be ordered, then send to
+//! `scan transaction`
+//! - When `scan transaction` returns succesffuly, remove txid from queued transactions
+//!
+//! On reload:
+//! Queued transactions list` will ensure that when wallet is saved that one of three cases
+//! will keep wallet state correct
```
Comment (fluidvanadium) [inline zingo-sync/src/scan_director.rs:43]: i have a slightly different way to implement this. will create a wallet save data specification

File: zingo-sync/src/scan_transaction.rs [hunk 4]
```diff
@@ -0,0 +1,10 @@
+//! Scans a transaction.
+//!
+//! - Takes a TXID from `scan director`
+//! - Requests the transaction from the `proxy connector`
+//! - Decrypts all the outputs and creates shielded notes
+//! - Creates transparent notes for the wallet's transparent addresses
+//! - Creates a basic transaction structure with references to all nullifiers and notes
+//! - Returns success result
+
+mod fees;
```
Comment (fluidvanadium) [inline zingo-sync/src/scan_transaction.rs:10]: why is this submod here? creating/sending transactions should probably be its own mod

**Review by fluidvanadium — COMMENTED** [review]
this is great. im going to write up something similar specifying what data i think we need to record

**Comment by Oscar-Pepper** [comment]
replaced by #829

**Pull request closed without merging**